src/wh_server.c

```diff
@@ -21,29 +21,35 @@
 #include "wolfhsm/wh_error.h"
 #include "wolfhsm/wh_comm.h"
 
-#if 0
-#include "wolfhsm/nvm.h"
-#include "wolfhsm/nvm_remote.h"
-#endif
+#include "wolfhsm/wh_nvm.h"
 
 #include "wolfhsm/wh_message.h"
 #include "wolfhsm/wh_message_comm.h"
+#include "wolfhsm/wh_message_nvm.h"
 #include "wolfhsm/wh_server.h"
 
 int wh_Server_Init(whServerContext* server, whServerConfig* config)
 {
     int rc = 0;
-    if (server == NULL) {
+    if ((server == NULL) || (config == NULL)) {
         return WH_ERROR_BADARGS;
     }
 
     memset(server, 0, sizeof(*server));
-    if (
-/*            ((rc = wh_Nvm_Init(server->nvm_device, config->nvm_device)) == 0) && */
-        ((rc = wh_CommServer_Init(server->comm, config->comm)) == 0) &&
-/*        ((rc = wh_NvmServer_Init(server->nvm, config->nvm)) == 0)*/
-        1) {
-        /* All good */
+
+    if (    (config->nvm_config->cb != NULL) &&
+            (config->nvm_config->cb->Init != NULL)) {
+
+        rc = config->nvm_config->cb->Init(config->nvm_config->context, config->nvm_config->config);
+    }
+    if (rc == 0) {
+        server->nvm->cb = config->nvm_config->cb;
+        server->nvm->context = config->nvm_config->context;
```

Comment:
Yeah, I re-added the helper layer in whNvm.c as it made the code much more readable because there had to be a number of parameter validations performed before invoking the cb.  We should probably do the same for comms as well to keep the pattern.

Reply:
cool. Meant to delete this comment and replace with the one two below, but clearly forgot

Reply:
I realize now this comment should have selected the line:
```
        rc = config->nvm_config->cb->Init(config->nvm_config->context, config->nvm_config->config);
```
not the context/callback settings. 


But now that we have the helper layer, should we use that here, vs invoking callbacks directly?